Source/WebCore/editing/TextListParser.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "TextListParser.h"
+
+#include <WebCore/CSSValueKeywords.h>
+#include <WebCore/FontAttributes.h>
+#include <span>
+#include <wtf/ASCIICType.h>
+#include <wtf/CheckedArithmetic.h>
+#include <wtf/StdLibExtras.h>
+#include <wtf/text/AtomString.h>
+#include <wtf/text/ParsingUtilities.h>
+#include <wtf/text/StringParsingBuffer.h>
+#include <wtf/unicode/CharacterNames.h>
+
+namespace WebCore {
+
+// MARK: Helpers
+
+template<typename Character>
+constexpr int consumeNumber(StringParsingBuffer<Character>& input)
+{
+    // Parse the digits until there is no more input left or a non-ASCII digit character has been encountered.
+    WTF::Checked<int> value;
+    do {
+        auto c = input.consume();
+        int digitValue = c - '0';
+        value = (value * 10) + digitValue;
+    } while (!input.atEnd() && WTF::isASCIIDigit(*input));
+
+    ASSERT(value.value() > 0);
+    return value.value();
+}
+
+// MARK: Primary consumers
+
+template<typename Character>
+std::optional<TextList> tryConsumeUnorderedDiscTextList(StringParsingBuffer<Character>& input)
+{
+    if (WTF::skipExactly(input, '*') ||  WTF::skipCharactersExactly(input, WTF::spanReinterpretCast<const Character>(WTF::span(WTF::Unicode::bullet)))) {
+        if (input.atEnd())
+            return { { { CSS::Keyword::Disc { } }, 0, false } };
+
+        // Invalid; skip to end.
```

Comment:
Instead of adding a comment like this, can we have a helper function with a descriptive name: e.g.
`skipToEndForInvalidInput(input)`

Reply:
This comment still withstands.

Reply:
ya, makes sense; I haven’t addressed all your comments yet, just trying to get it to build in EWS atm!